website/common/script/ops/scoreTask.js

```diff
@@ -130,7 +130,7 @@ function _addPoints (user, task, stats, direction, delta) {
 
     stats.gp += afterStreak;
   } else {
-    stats.gp += gpMod;
+    stats.gp = parseInt(stats.gp) + gpMod;
```

Comment:
this change is made into scoreTask.js, if I read correctly the issue is about the Fix Character Values. The code for that modal can be found here https://github.com/HabitRPG/habitica/blob/develop/website/client/components/settings/resetModal.vue

Reply:
Yes but the core problem is in scoreTask because the real problem is not in the Settings component but in the Task component when you click the positive button in a Habit.
Then a method called 'score' is called which calls scoreTask.js.

Reply:
I've never heard of problems when scoring an habit and I'm sure we would have got a lot of reports if this was the case. Also the settings component doesn't use scoreTask to adjust the gold points. The problem here is that probably the number is treated as  string so doing numb += numb append the new value to the string

Reply:
I'm new to this repo, but the changes I made fix the error. I don't  know why you say that if that was the case it will 'cause more problems, I might have to investigate more on that. But the append thing is actually what's happening. In scoreTask.js the method in which I made the change is using a 'stats' object which has 'gp' and the value is a string, that's why the parseInt() fixes the problem.

Here's a gift showing the actual problem.
![habitica-error](https://user-images.githubusercontent.com/1399744/31787320-36eb2bd6-b4c0-11e7-8574-66861d7cacb4.gif)


And here's one with the same steps but applying my changes.
![habitica-fix](https://user-images.githubusercontent.com/1399744/31787591-15d8704c-b4c1-11e7-9ebd-4bedc55cbb51.gif)


Can you please guide me on why that's not the fix and what can I do to actually contribute?